app/src/main/java/com/dessalines/thumbkey/db/AppDb.kt

```diff
@@ -613,8 +635,44 @@ val MIGRATION_16_17 =
         }
     }
 
+val MIGRATION_17_18 =
+    object : Migration(17, 18) {
+        override fun migrate(db: SupportSQLiteDatabase) {
+            db.execSQL(
+                "ALTER TABLE AppSettings RENAME COLUMN key_size TO key_size_defunct",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings RENAME COLUMN key_width TO key_width_defunct",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN auto_size_keys INTEGER NOT NULL DEFAULT $DEFAULT_AUTO_SIZE_KEYS",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN non_square_keys INTEGER NOT NULL DEFAULT $DEFAULT_NON_SQUARE_KEYS",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN key_width_v18 INTEGER NOT NULL DEFAULT $DEFAULT_KEY_WIDTH",
+            )
+            db.execSQL(
+                "ALTER TABLE AppSettings ADD COLUMN key_height_v18 INTEGER NOT NULL DEFAULT $DEFAULT_KEY_HEIGHT",
```

Comment:
I believe all these can be combined like:

```sql
alter table AppSettings
  add column ...,
  add column ...,
   ...
```

If not, this works.

Reply:
So apparently SQLite can handle this (without the commas) but Room chokes. It crashed on me with a relevant(ish) error and I found some SO comments about it not being possible.

Reply:
Gotcha, no probs.